src/internal/starlark/starlark.go

```diff
@@ -0,0 +1,232 @@
+// Package starlark runs Pachyderm-specific starlark programs.  It's recommended to import this as
+// "ourstar" when also using go.starlark.net/starlark in the same package.
+package starlark
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/log"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+	starjson "go.starlark.net/lib/json"
+	starmath "go.starlark.net/lib/math"
+	startime "go.starlark.net/lib/time"
+	"go.starlark.net/starlark"
+	"go.starlark.net/syntax"
+	"go.uber.org/zap"
+)
+
+const (
+	goContextKey       = "goContext"
+	scriptDirectoryKey = "scriptDirectory"
+)
+
+func init() {
+	starlark.Universe["json"] = starjson.Module
+	starlark.Universe["time"] = startime.Module
+	starlark.Universe["math"] = starmath.Module
+}
+
+// GetContext returns a context.Context that applies to in the provided Starlark thread.
+func GetContext(t *starlark.Thread) context.Context {
+	x := t.Local(goContextKey)
+	if ctx, ok := x.(context.Context); ok {
+		return ctx
+	}
+	return pctx.TODO()
+}
+
+func scriptDirectory(t *starlark.Thread) string {
+	x := t.Local(scriptDirectoryKey)
+	if s, ok := x.(string); ok {
+		return s
+	}
+	return "/dev/null"
+}
+
+func nameScript(t *starlark.Thread, abs string) string {
+	wd, err := os.Getwd()
+	if err != nil {
```

Comment:
On Linux:

```
       EACCES Permission to read or search a component of the filename was denied.

       EFAULT buf points to a bad address.

       EINVAL The size argument is zero and buf is not a null pointer.

       EINVAL getwd(): buf is NULL.

       ENAMETOOLONG
              getwd(): The size of the null-terminated absolute pathname string exceeds PATH_MAX bytes.

       ENOENT The current working directory has been unlinked.

       ENOMEM Out of memory.

       ERANGE The  size argument is less than the length of the absolute pathname of the working directory, including the terminating
              null byte.  You need to allocate a bigger array and try again.
```

These are unlikely, but all this does is name the file in stack traces.  So if you start a program running and delete your current directory before module loading happens, then error messages will be `/home/you/pach/pachyderm/scripts/foo.star:123: syntax error` instead of `scripts/foo.start:123: syntax error`.  It is harmless, I think.